### ISSUE TITLE
Significant drop in model accuracy with hist method

### ISSUE THREAD

**Issue by hcho3** [dmlc/xgboost#7783]
https://discuss.xgboost.ai/t/xgboost-xgboost-hist-and-lightgbm-benchmark-xgboost-hist-performance-drop/2719

**Comment by trivialfis** [comment]
The benchmark is using 1.5.2, will try to reproduce it later.

**Comment by trivialfis** [comment]
@hcho3 You fixed it in https://github.com/dmlc/xgboost/pull/7551 .
